druid/src/widget/flex.rs

```diff
@@ -583,39 +574,34 @@ impl<T: Data> Flex<T> {
     ///
     /// [`add_default_spacer`]: #method.add_default_spacer
     pub fn add_spacer(&mut self, len: impl Into<KeyOrValue<f64>>) {
-        let spacer = Spacer {
-            axis: self.direction,
-            len: len.into(),
-        };
-        self.add_flex_child(spacer, 0.0);
+        let value = len.into();
+        let new_child = Child::FixedSpacer(value, 0.0);
+        self.children.push(new_child);
     }
 
     /// Add an empty spacer widget with a specific `flex` factor.
     pub fn add_flex_spacer(&mut self, flex: f64) {
-        let child = match self.direction {
-            Axis::Vertical => SizedBox::empty().expand_height(),
-            Axis::Horizontal => SizedBox::empty().expand_width(),
-        };
-        self.add_flex_child(child, flex);
+        let new_child = Child::FlexedSpacer(flex, 0.0);
+        self.children.push(new_child);
     }
 }
 
 impl<T: Data> Widget<T> for Flex<T> {
     fn event(&mut self, ctx: &mut EventCtx, event: &Event, data: &mut T, env: &Env) {
-        for child in &mut self.children {
-            child.widget.event(ctx, event, data, env);
+        for child in self.children.iter_mut().filter_map(|x| x.widget_mut()) {
+            child.event(ctx, event, data, env)
```

Comment:
doesn't matter except that it can be slightly visually confusing.

```suggestion
            child.event(ctx, event, data, env);
```

Reply:
weird that it removed it?? I didnt change anything of the lines besides pasteing child in place of widget. 
VScode also changed indentation, so maybe it also removed the `;` for me.